pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -1362,6 +1363,37 @@ func (ss stepSelector) Filter(predicate func(*singleStep) bool) stepSelector {
 	return result
 }
 
+// Cut returns the before and after of a cut step selector, as well as the step that was cut out.
+func (ss stepSelector) Cut(
+	predicate func(*singleStep) bool,
+) (stepSelector, stepSelector, stepSelector) {
+	var before, after, cut stepSelector
+	predicateFound := false
+	for _, s := range ss {
+		if predicateFound {
+			after = append(after, s)
+		} else if predicate(s) {
+			predicateFound = true
+			cut = append(cut, s)
+		} else {
+			before = append(before, s)
+		}
+	}
+	return before, after, cut
+}
+
+// CutAfter cuts the selector and returns the two halves with the after including the cut step.
```

Comment:
nit: for functions like this you can refer back to the base function, i.e. `Cut` to ease the explanation/reduce duplicate comments. e.g:

```
// CutAfter is like `Cut` except it merges the `cut` step with the `after` selector.
```